src/index.ts

```diff
@@ -447,14 +448,38 @@ class WorkerInfo extends AsynchronouslyCreatedResource {
     )
   }
 
-  async destroy(): Promise<void> {
-    await this.worker.terminate()
-    this.port.close()
-    this.clearIdleTimeout()
-    for (const taskInfo of this.taskInfos.values()) {
-      taskInfo.done(Errors.ThreadTermination())
-    }
-    this.taskInfos.clear()
+  async destroy(timeout?: number): Promise<void> {
+    let resolve: () => void
+    let reject: (err: Error) => void
+
+    const ret = new Promise<void>((res, rej) => {
+      resolve = res
+      reject = rej
+    })
+
+    const timer = timeout
+      ? setTimeout(
+          () => reject(new Error('Failed to terminate worker')),
```

Comment:
One idea I had was that here we could tell the worker to `process.exit` if main thread's `worker.terminate` was taking too long or got stuck.

```ts
setTimeout(() => {
  this.worker.postMessage('TERMINATE_NOW')
  reject(new Error('Failed to terminate worker')
})
```

Worker would catch this:

```ts
// worker.ts
parentPort.on('message', (message) => {
  // Main thread tried to terminate this worker but failed, let's force the exit here in worker
  if (message === 'TERMINATE_NOW') {
    process.exit()
  }
})
```

But it seems that this does not work. The worker never receives the message. I guess the `worker.terminate()` has already closed the message channel even though it has not yet resolved completely.